src/artemis/external_interaction/nexus/write_nexus.py

```diff
@@ -285,6 +121,117 @@ def update_nexus_file_timestamp(self):
             shutil.copy(filename, temp_filename)
             with h5py.File(temp_filename, "r+") as nxsfile:
                 nxsfile["entry"].create_dataset(
-                    "end_time", data=np.string_(self._get_current_time())
+                    "end_time", data=np.string_(get_current_time())
                 )
             shutil.move(temp_filename, filename)
+
+    def get_image_datafiles(self, max_images_per_file=1000):
+        return [
+            self.directory / f"{self.filename}_{h5_num + 1:06}.h5"
+            for h5_num in range(-(-self.full_num_of_images // max_images_per_file))
+        ]
+
+
+class FGSNexusWriter(NexusWriter):
+    grid_scan: dict
+
+    def __init__(self, parameters: FGSInternalParameters, grid_scan: dict) -> None:
+        super().__init__(parameters)
+        self.goniometer = create_gridscan_goniometer_axes(
```

Comment:
Should: I think it would be nicer to have an abstract function to get the gonio that each subclass needs to implement. It's more explicitly obvious it's needed then

Reply:
I still think this

Reply:
will do this in the other branch